### ISSUE TITLE
Option to disable automatic page break insertion

### ISSUE THREAD

**Issue by matklad** [asciidoctor/asciidoctor#2972]
Hi!

I'd love to use `asciidoctor-pdf` to produce presentations. I am aware of ascidoctor-reveal.js, but PDF's just work better for me: ideally, I'd love "beamer with asciidoctor syntax". 

The main blocker I see at the moment is that asciidoctor-pdf automatically inserts pagebreaks. For slides, I really want to control page breaks manually and deal with `Prawn::Errors::CannotFit` errors. 

It would be cool to have an attribute to disable this behavior globally per-document. https://github.com/asciidoctor/asciidoctor-pdf/issues/38 is a related, but distinct issue.

**Comment by mojavelinux** [comment]
I'm not really sure what you mean by "automatically inserts pagebreaks". Once the number of lines on a page are exhausted, the PDF library (Prawn) advances to the next page. The goal of Asciidoctor PDF is to create a flow-oriented PDF document. And that is unlikely to change.

It sounds to me like you're trying to use the wrong technology to solve your problem. You might be much better off using the print support in Asciidoctor Reveal.js or Asciidoctor Bespoke. Those converters are designed specifically for creating slides and laying them out as a PDF.

To print slides with Asciidoctor Bespoke, see https://github.com/opendevise/presentation-bespoke-starter

**Comment by matklad** [comment]
>It sounds to me like you're trying to use the wrong technology to solve your problem.

Probably yeah, feel free to close in this case. 

Still, `adoc -> HTML -> PDF` pipeline does not feel exactly right to me, I'd prefer to produce `PDF` directly. But that could very well mean implementing a separate back end for ascidoctor, oriented for PDFs with more manual layout control.

**Comment by mojavelinux** [comment]
I'd argue it's exactly the right pipeline. In fact, we're going to be going that way with our PDF strategy in general. Creating a layout engine for PDF is an enormous undertaking, as I have learned. It's much smarter to leverage the incredible layout capabilities offered by the browser and simply distill that to PDF. That's partly because of the maturity of those technologies (specifically CSS), but also because PDF was simply not designed for this. There's just no equivalent mechanism in PDF, so you end up having to do a ton of manual layout, and that has serious limits (as you can see by browsing the issues in this repository).

**Comment by matklad** [comment]
Thanks a lot for the input, this totally makes sense! Will try to get HTML -> PDF part working then!

**Comment by mojavelinux** [comment]
You might want to check out [asciidoctor-pdf.js](https://github.com/Mogztter/asciidoctor-pdf.js). That's the new effort.

**Comment by matklad** [comment]
Ohh, that looks awesome, thanks a lot for that link!

**Comment by obilodeau** [comment]
> Will try to get HTML -> PDF part working then!

I use decktape for this: https://github.com/astefanutti/decktape. Reveal.js had bugs but I got it to work with a recent version without using decktape. You need to add `?print-pdf=1` or something similar to the URL and I think it only works with chrome-based.

That said, I'm still sticking to decktape because it works from the command-line so it's easier to streamline your slides build.

**Comment by matklad** [comment]
Reveal.js/remark/slides.com didn’t work for me quiet as well. However asdiidoctor-pdf.js is *perfect*, because I have full control over HTML, and so can write just a bare minimum of JS/CSS to produce a PDF. 

Here’s what I did yesterday with it: 
https://github.com/matklad/pytesting-slides/blob/master/out/slides.pdf

**Comment by obilodeau** [comment]
Yeah after replying I took a look at asciidoctor-pdf.js since I struggled with PDF generation for papers with asciidoctor-pdf and seeing Dan's vision. I have to say I'm really impressed with what was done with very little time!

Cool slides, minimal markup, I love it! I'm definitely going to borrow your `two-col` CSS as an asciidoctor-reveal.js built-in, thanks for that :smile:

**Comment by mojavelinux** [comment]
> I struggled with PDF generation for papers with asciidoctor-pdf and seeing Dan's vision.

I no longer see my own vision with Asciidoctor PDF. There's no doubt in my mind that the way to get the PDFs we want is to leverage the browser to handle the style and layout. The possibilities are limitless. (I should have realized this right after creating Asciidoctor Bespoke, which gives me that outcome for the same reason).
